src/context.rs

```diff
@@ -127,64 +127,64 @@ impl Context {
     #[allow(clippy::too_many_arguments)]
     pub fn start_auth_session(
         &mut self,
-        tpm_key: ESYS_TR,
+        tpm_key: Option<KeyHandle>,
         bind: ESYS_TR,
         nonce: Option<&Nonce>,
         session_type: SessionType,
         symmetric: Cipher,
         auth_hash: HashingAlgorithm,
-    ) -> Result<Session> {
+    ) -> Result<Option<Session>> {
         let nonce_ptr: *const TPM2B_NONCE = match nonce {
             Some(val) => &TPM2B_NONCE::try_from(val.clone())?,
             None => null(),
         };
 
-        let mut session_handle = ESYS_TR_NONE;
+        let mut esys_session_handle = ESYS_TR_NONE;
 
         let ret = unsafe {
             Esys_StartAuthSession(
                 self.mut_context(),
-                tpm_key,
+                tpm_key.map(|v| v.into()).unwrap_or(ESYS_TR_NONE),
                 bind,
-                self.sessions.0,
-                self.sessions.1,
-                self.sessions.2,
+                Session::handle(self.sessions.0).into(),
```

Comment:
Maybe I'm a bit late on this one 🤡  you could implement 3 methods on context:
```Rust
    fn session_0(&self) -> ESYS_TR {
        Session::handle(self.sessions.0).into()
    }
```
and use that everywhere, like we use the `self.mut_context()`

Reply:
Yes! That is much better. I think I will remove the ```From``` implementations. Because I do not like them and just keep static handle function that takes ```Option<Session>``` and maybe add another method to Session that retrieves can retrieve the handle.